device/samples/pnp/pnpTemperatureController.js

```diff
@@ -166,7 +172,32 @@ async function sendTelemetry(deviceClient, data, index, componentName) {
   await deviceClient.sendEvent(pnpMsg);
 }
 
+async function provisionDevice(payload) {
+  var provSecurityClient = new SymmetricKeySecurityClient(registrationId, symmetricKey);
+  var provisioningClient = ProvisioningDeviceClient.create(provisioningHost, idScope, new ProvProtocol(), provSecurityClient);
+
+  if (!!(payload)) {
+    provisioningClient.setProvisioningPayload(payload);
+  }
+
+  try {
+    let result = await provisioningClient.register();
```

Comment:
you need a switch here which if dps u do this otherwise we go normally to connect to hub with connection string..